### DISCUSSION TITLE
Align images so that you can better compare them

### DISCUSSION THREAD

**Discussion by eddydm57** [ImageMagick/ImageMagick#7269] [Digital Image Processing]
I'm new in using imagemagick.
I'm also learning to draw. 
I'm already used to scan my drawings in different stages of the devellopment.
Now I would like to use imagemagick to compare the different stages.

But I have already detected that after the scan-action, the jpg-files are little different, perhaps because i didnt put the original in identical the same way on the scanner.
Is there a methode to find some 'marker' in a jpg file so that i can align that marker in all the pictures on the same place? and than use the compare action? ( For example: i can start each drawing with a small cross in the upper left corner of the drawing as a marker and leave it there during the whole process.)

Thanks in advance.
Eddy

**Comment by snibgo** [comment]
Yes.

If you align with a single mark, this defines a translation, a movement in the x- and y-directions. Aligning with two marks (eg in opposite corners) defines a translation and rotation and scale. If you use the same physical scanner, the scale probably doesn't change. Three points also allows correction for shear.

To do this, scan your drawing with one or more marks. Find the marks, and make crops of them. Keep these crops, and make a note of their locations. For example, the top-left corner of one crop might be at coordinates 123,156. This could be a manual process, or easily automated if the drawing contains nothing but the marks.

When the drawing has progressed, scan it again. Find the marks on this scan. You can do this by searching it for each of the saved crops. (`magick compare ... -subimage-search`) If you scan in the same way, the marks won't have moved much, so you don't need to search the entire scan for them. For example, the mark might now be at coordinates 111,167.

Then you can make a version of the second scan that matches the first like this:
```
magick second.png -distort Affine 111,167,123,156 corrected.png
```
If you have 2 or 3 marks, use another 4 or 8 numbers.

**Reply by eddydm57** [reply]
Thanks for this answer!
Do you see a possibility to export te result of the magick compage ... -subimage-search for example in a txt-file so that i can in a program read those coordinaties?

**Comment by fmw42** [comment]
This works on my Mac OSX Ventura and should work on Windows also

`magick compare -metric rmse image1.suffix image2.suffix null: 2> compare.txt` 

You can name the resulting txt file as you want.

**Reply by eddydm57** [reply]
Thanks for this answer.
The possibilities of magick are so big, so it is impossible to learn all the possibilities in 1-2-3
